pkg/cache/pubsub.go

```diff
@@ -118,54 +140,55 @@ func (c *cachePubSub) PublishToTopicWithSecondaryKey(secondaryKey string, data i
 	}
 
 	// Data has changed
-	globalCache.Set(c.topic, data)
-	globalCache.SetSecondaryKey(c.topic, secondaryKey)
-	c.channel <- data
-	return nil
+	return c.setAndSendWithSecondaryKey(c.topic, secondaryKey, data)
 }
 
 // PublishCacheHash - send in data to publish, if it has changed update cache, storing only the hash, notify subscribers
-func (c *cachePubSub) PublishCacheHash(key, secondarykey string, data interface{}) error {
+func (c *cachePubSub) PublishCacheHash(key, secondaryKey string, data interface{}) error {
 	hash, err := util.ComputeHash(data)
+	if err != nil {
+		return err
+	}
+
 	changed, err := c.hasChanged(key, hash)
 	if !changed || err != nil {
 		return err
```

Comment:
could probably still combine all of these lines into a func that returns err, hash, and changed (maybe don't even need changed) because these lines are all repeated multiple x.

Reply:
They are repeated because I don't want to update the cache, or send it on the channel, if the item has not changed.  I combined what I could to get this, but still would require a return and if check if I moved this in anyway.

Reply:
ok